### ISSUE TITLE
Unification scripting tests are non-deterministically failing

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#6588]
References_Versioning_FxUnification2 and References_Versioning_FxUnification1 are failing on my box with:

```
xUnit.net Console Runner (32-bit .NET 4.0.30319.42000)
  Discovering: Microsoft.CodeAnalysis.CSharp.Scripting.Desktop.UnitTests
  Discovered:  Microsoft.CodeAnalysis.CSharp.Scripting.Desktop.UnitTests
  Starting:    Microsoft.CodeAnalysis.CSharp.Scripting.Desktop.UnitTests
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.CsiTests.ReferenceSearchPaths_Sdk
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.DebuggerProxy_FrameworkTypes_SortedList
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References2
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.DebuggerProxy_FrameworkTypes_Queue
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.DebuggerProxy_FrameworkTypes_Stack
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.VBBackingFields_DebuggerBrowsable
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.DebuggerProxy_FrameworkTypes_ArrayList
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.DebuggerProxy_FrameworkTypes_Hashtable
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_FxUnification2
      
      Actual:
      "mscorlib, Version=4.0.0.0"
      Differences:
      --> "System, Version=2.0.0.0",
          "mscorlib, Version=4.0.0.0",
      --> "System.Configuration, Version=2.0.0.0: <implicit>,global",
      --> "System.Xml, Version=2.0.0.0: <implicit>,global",
      --> "System.Data.SqlXml, Version=2.0.0.0: <implicit>,global",
      --> "System.Security, Version=2.0.0.0: <implicit>,global"
      
      Expected: True
      Actual:   False
      Stack Trace:
        src\Test\Utilities\Shared\Assert\AssertEx.cs(209,0): at Roslyn.Test.Utilities.AssertEx.Equal[T](IEnumerable`1 expected, IEnumerable`1 actual, IEqualityComparer`1 comparer, String message, String itemSeparator, Func`2 itemInspector)
        src\Test\Utilities\Shared\Compilation\CompilationExtensions.cs(142,0): at Microsoft.CodeAnalysis.Test.Utilities.CompilationExtensions.VerifyAssemblyVersionsAndAliases(Compilation compilation, String[] expectedAssembliesAndAliases)
        src\Scripting\CSharpTest.Desktop\InteractiveSessionTests.cs(272,0): at Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_FxUnification2()
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.SearchPaths_BaseDirectory
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.MissingRefrencesAutoResolution
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.AssemblyResolution
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.CompilationChain_UsingRebinding_Directive
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.CompilationChain_SubmissionSlots
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_WeakNames3
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.HostObjectBinding_DuplicateReferences
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.SearchPaths_RemoveDefault
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_FxUnification1
      
      Actual:
      "System, Version=4.0.0.0",
      "mscorlib, Version=4.0.0.0",
      "System.Configuration, Version=4.0.0.0: <implicit>,global",
      "System.Xml, Version=4.0.0.0: <implicit>,global",
      "System.Data.SqlXml, Version=4.0.0.0: <implicit>,global",
      "System.Security, Version=4.0.0.0: <implicit>,global",
      "System.Core, Version=4.0.0.0: <implicit>,global",
      "System.Numerics, Version=4.0.0.0: <implicit>,global"
      Differences:
      --> "System, Version=2.0.0.0: <superseded>",
          "System, Version=4.0.0.0",
          "mscorlib, Version=4.0.0.0",
          "System.Configuration, Version=4.0.0.0: <implicit>,global",
          "System.Xml, Version=4.0.0.0: <implicit>,global",
          "System.Data.SqlXml, Version=4.0.0.0: <implicit>,global",
          "System.Security, Version=4.0.0.0: <implicit>,global",
          "System.Core, Version=4.0.0.0: <implicit>,global",
          "System.Numerics, Version=4.0.0.0: <implicit>,global",
      --> "System.Configuration, Version=2.0.0.0: <superseded>",
      --> "System.Xml, Version=2.0.0.0: <superseded>",
      --> "System.Data.SqlXml, Version=2.0.0.0: <superseded>",
      --> "System.Security, Version=2.0.0.0: <superseded>"
      
      Expected: True
      Actual:   False
      Stack Trace:
        src\Test\Utilities\Shared\Assert\AssertEx.cs(209,0): at Roslyn.Test.Utilities.AssertEx.Equal[T](IEnumerable`1 expected, IEnumerable`1 actual, IEqualityComparer`1 comparer, String message, String itemSeparator, Func`2 itemInspector)
        src\Test\Utilities\Shared\Compilation\CompilationExtensions.cs(142,0): at Microsoft.CodeAnalysis.Test.Utilities.CompilationExtensions.VerifyAssemblyVersionsAndAliases(Compilation compilation, String[] expectedAssembliesAndAliases)
        src\Scripting\CSharpTest.Desktop\InteractiveSessionTests.cs(242,0): at Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_FxUnification1()
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.SearchPaths1
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References1
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References1 [SKIP]
      xunit2
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_StrongNames1
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_StrongNames2
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.CompilationChain_GlobalImportsRebinding
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.CompilationChain_UsingRebinding_AddReference
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.HostObjectInInMemoryAssembly
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_WeakNames2
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.References_Versioning_WeakNames1
    Microsoft.CodeAnalysis.CSharp.Scripting.Test.InteractiveSessionTests.ReferenceToInvalidType
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.CsiTests.CurrentWorkingDirectory1
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.CsiTests.ReferenceSearchPaths_LIB
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.CsiTests.CurrentWorkingDirectory_Change
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.CsiTests.CurrentWorkingDirectory_Change [SKIP]
      https://github.com/dotnet/roslyn/issues/6523
    Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.CsiTests.DefaultUsings
  Finished:    Microsoft.CodeAnalysis.CSharp.Scripting.Desktop.UnitTests
=== TEST EXECUTION SUMMARY ===
   Microsoft.CodeAnalysis.CSharp.Scripting.Desktop.UnitTests  Total: 32, Errors: 0, Failed: 2, Skipped: 2, Time: 41.018s

```

**Comment by davkean** [comment]
tag @tmat

**Comment by tmat** [comment]
@davkean Is .NET FX 3.5 not installed on the box?

**Comment by davkean** [comment]
Probably not, why would it be?

**Comment by tmat** [comment]
So I assume the test is failing consistently?

**Comment by davkean** [comment]
Failing consistently on one of my boxes - but not on other boxes, hence the "non-deterministic" title. :)

**Comment by tmat** [comment]
The other box has .NET 3.5 installed, right?

**Comment by tmat** [comment]
Fixed by #6740
